game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProPurchaseOption.java

```diff
@@ -72,7 +72,7 @@ public class ProPurchaseOption {
     this.unitType = unitType;
     this.player = player;
     final UnitAttachment unitAttachment = unitType.getUnitAttachment();
-    final Resource pus = data.getResourceList().getResource(Constants.PUS);
+    final Resource pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.